raster/src/main/scala/geotrellis/raster/io/geotiff/GeoTiff.scala

```diff
@@ -52,6 +52,21 @@ trait GeoTiff[T <: CellGrid] extends GeoTiffData {
  * Companion object to GeoTiff
  */
 object GeoTiff {
+  def readMultiband(path: String): MultibandGeoTiff =
+    MultibandGeoTiff(path)
+
+  def readSingleband(path: String): SinglebandGeoTiff =
+    SinglebandGeoTiff(path)
+
+  def apply(path: String): Either[SinglebandGeoTiff, MultibandGeoTiff] = {
+    val multiband = MultibandGeoTiff(path)
+    if (multiband.tile.bandCount == 1) {
+      Left(SinglebandGeoTiff(path))
```

Comment:
You don't actually need to read tiff from the source path again as you already read geotiff into memory using `MultibanfGeotiff(path)`:

```scala
new SinglebandGeoTiff(tile = multiband.tile.band(0), multiband.extent, multiband.crs, multiband.tags, multiband.options)
```